flows/contact.go

```diff
@@ -169,7 +169,7 @@ func (c *Contact) ResolveQueryKey(key string) []interface{} {
 		urnsWithScheme := c.urns.WithScheme(key)
 		vals := make([]interface{}, len(urnsWithScheme))
 		for u := range urnsWithScheme {
-			vals[u] = string(urnsWithScheme[u])
+			vals[u] = string(urnsWithScheme[u].URN)
```

Comment:
Shouldn't this be returning a list of `ContactURN` objects for the scheme? IE, shouldn't this kind of scheme filtering be returning a filtered list of the same type as the main URN list? Or am I understanding this wrong?

Reply:
So the practical reason is - we have three places URNs occur in goflow. 1) on contacts 2) events 3) actions. Contacts are unmarshaled in the context of a session. They have a separate envelope struct and real struct and they can resolve things like channels as they are unmarshalled. Actions and events are somewhat simpler beasts - they are unmarshaled straight into their struct without access to the session. 

Viewed one way that's a bit inconsistent. Viewed another way that's a nice optimization. A session might have a lot of old events and it's nice to keep their unmarshaling relatively lightweight. It also means we can parse a session with events that reference assets which are no longer available. For example an old event might reference a channel which has been since deleted - we don't want to blow up in that case because we can't fetch the channel.